src/frontend/XAPI/index.ts

```diff
@@ -0,0 +1,94 @@
+// https://liveaspankaj.gitbooks.io/xapi-video-profile/content/statement_data_model.html
+
+import uuid from 'uuid';
+import { XAPI_ENDPOINT } from '../settings';
+
+const ContextExtensionsDefintion: { [key: string]: string } = {
+  ccSubtitleEnabled:
+    'https://w3id.org/xapi/video/extensions/cc-subtitle-enabled',
+  ccSubtitleLanguage: 'https://w3id.org/xapi/video/extensions/cc-subtitle-lang',
+  completionTreshold:
+    'https://w3id.org/xapi/video/extensions/completion-threshold',
+  frameRate: 'https://w3id.org/xapi/video/extensions/frame-rate',
+  fullScreen: 'https://w3id.org/xapi/video/extensions/full-screen',
+  length: 'https://w3id.org/xapi/video/extensions/length',
+  quality: 'https://w3id.org/xapi/video/extensions/quality',
+  screenSize: 'https://w3id.org/xapi/video/extensions/screen-size',
+  sessionId: 'https://w3id.org/xapi/video/extensions/session-id',
+  speed: 'https://w3id.org/xapi/video/extensions/speed',
+  track: 'https://w3id.org/xapi/video/extensions/track',
+  userAgent: 'https://w3id.org/xapi/video/extensions/user-agent',
+  videoPlaybackSize:
+    'https://w3id.org/xapi/video/extensions/video-playback-size',
+  volume: 'https://w3id.org/xapi/video/extensions/volume',
+};
+
+const ResultExtensionsDefinition: { [key: string]: string } = {
+  playedSegment: 'https://w3id.org/xapi/video/extensions/played-segments',
+  progress: 'https://w3id.og/xapi/video/extensions/progress',
+  time: 'https://w3id.org/xapi/video/extensions/time',
+  timeFrom: 'https://w3id.org/xapi/video/extensions/time-from',
+  timeTo: 'https://w3id.org/xapi/video/extensions/time-to',
+};
+
+export const verbDefinition: { [key: string]: string } = {
```

Comment:
Maybe we could add some docstrings, at least on exported members? It has not always been done but I always do it these days and I think it's pretty helpful to navigate the codebase